docs/frontend-performance.md

```diff
@@ -0,0 +1,2984 @@
+        <TransactionItem key={tx.hash} transaction={tx} />
+      ))}
+      {hasMore && (
+        <button onClick={loadMore} disabled={isLoading}>
+          Load More
+        </button>
+      )}
+    </div>
+  );
+};
+```
+
+```typescript
+❌ WRONG: Load all assets at once
+const AssetList = ({ accountId }: AssetListProps) => {
+  const [assets, setAssets] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    // Loads ALL assets at once - blocks UI
+    fetchAllAssets(accountId).then(allAssets => {
+      setAssets(allAssets); // 1000+ assets loaded at once!
+      setLoading(false);
+    });
+  }, [accountId]);
+
+  return loading ? <Spinner /> : <div>{assets.map(a => <Asset key={a.id} asset={a} />}</div>;
+};
+```
+
+```typescript
+✅ CORRECT: Progressive pagination
+const AssetList = ({ accountId }: AssetListProps) => {
+  const [assets, setAssets] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+  const [loading, setLoading] = useState(false);
+  const pageRef = useRef(0);
+  const PAGE_SIZE = 50;
+
+  const loadPage = useCallback(async () => {
+    if (loading || !hasMore) return;
+
+    setLoading(true);
+    try {
+      const result = await fetchAssetsPage(accountId, pageRef.current, PAGE_SIZE);
+      setAssets(prev => [...prev, ...result.items]);
+      setHasMore(result.hasMore);
+      pageRef.current += 1;
+    } finally {
+      setLoading(false);
+    }
+  }, [accountId, loading, hasMore]);
+
+  useEffect(() => {
+    // Load first page on mount
+    loadPage();
+  }, [accountId]); // Reset on account change
+
+  return (
+    <div>
+      {assets.map(a => <Asset key={a.id} asset={a} />)}
+      {hasMore && (
+        <button onClick={loadPage} disabled={loading}>
+          {loading ? 'Loading...' : 'Load More'}
+        </button>
+      )}
+    </div>
+  );
+};
+```
+
+---
+
+## Code Splitting and Lazy Loading
+
+### Use React.lazy for Route-Based Splitting
+
+```typescript
+❌ WRONG: Import all pages upfront
+import Settings from './pages/Settings';
+import Tokens from './pages/Tokens';
+import Activity from './pages/Activity';
+
+const App = () => {
+  return (
+    <Routes>
+      <Route path="/settings" element={<Settings />} />
+      <Route path="/tokens" element={<Tokens />} />
+      <Route path="/activity" element={<Activity />} />
+    </Routes>
+  );
+};
+
+✅ CORRECT: Lazy load pages
+import { lazy, Suspense } from 'react';
+
+const Settings = lazy(() => import('./pages/Settings'));
+const Tokens = lazy(() => import('./pages/Tokens'));
+const Activity = lazy(() => import('./pages/Activity'));
+
+const App = () => {
+  return (
+    <Suspense fallback={<LoadingSpinner />}>
+      <Routes>
+        <Route path="/settings" element={<Settings />} />
+        <Route path="/tokens" element={<Tokens />} />
+        <Route path="/activity" element={<Activity />} />
+      </Routes>
+    </Suspense>
+  );
+};
+```
+
+### Lazy Load Heavy Components
+
+```typescript
```

Comment:
When these React component examples are provided with JSX, we should probably use `jsx` syntax for markdown.

Instead of having:
```text
```typescript
```

We can use:
```text
```jsx
```

Then it will properly identify the JSX components and paint them in proper colors. It looks like in code editor then, and it's easier to read or identify specific code elements.

I think it properly keeps the TypeScript syntax coloring as well. So, no negative effect on it, I believe.